### PR TITLE
fix(mitm-addon): harden logging writes

### LINKED ISSUE #14371: test(mitm-addon): logging_utils write functions lack failure-path coverage
## Problem

`crates/runner/mitm-addon/src/logging_utils.py` ships two write functions whose error-handling contract is the failure-mode that matters most (a logger that crashes the proxy is worse than a logger that silently drops a line):

- `log_network_entry` (lines 14-25)
- `log_proxy_entry` (lines 28-45)

Both wrap the `json.dumps + os.open/write/close` chain in `except Exception as e: ctx.log.warn(...)`. The intent is clearly *swallow everything, never propagate to the proxy hook*. But existing tests only cover the happy paths and the empty-path no-op:

- `tests/test_utils.py:45-65` covers `log_proxy_entry` JSONL write, append, empty-path no-op.
- `tests/test_registry.py:441-467` covers `log_network_entry` JSONL write, append, empty-path no-op.

No test forces the `except Exception` branch to run, so a future change that breaks the swallow-and-log contract (e.g. changing the catch to a narrower class, or moving the `json.dumps` outside the try) would not be caught by CI.

Concrete uncovered failure modes:

1. **`os.open` raises `FileNotFoundError`** — passing `log_path='/nonexistent/dir/x.jsonl'` triggers it. Should not propagate.
2. **`os.open` raises `IsADirectoryError`** — passing a directory path. Should not propagate.
3. **`json.dumps` raises `TypeError`** — passing `**extra` with a non-serializable value (e.g. `bytes`, `set`, `datetime`) or an `entry` dict containing such a value. Currently caught by `except Exception` — but no test pins the behavior. Worth a regression test because mitm addon code does forward arbitrary metadata into the log (`add_firewall_metadata` copies `flow.metadata` keys; if a future caller passes a non-serializable value, current behavior is 'log line dropped', changing that silently to 'proxy hook crashes' would be bad).
4. **`os.write` raises `OSError` (e.g. ENOSPC, EIO)** — harder to trigger reliably in tests but the FileNotFoundError case exercises the same `except` handler.

## Suggested tests

Add to `tests/test_utils.py` (for `log_proxy_entry`) and `tests/test_registry.py` (or a new `TestLogNetworkEntryErrors` block):

```python
def test_log_proxy_entry_swallows_oserror(tmp_path):
    with patch.object(logging_utils.ctx, 'log', MagicMock(), create=True):
        # Parent dir does not exist — os.open raises FileNotFoundError.
        log_proxy_entry(str(tmp_path / 'nope' / 'p.jsonl'), 'warn', 'm')

def test_log_proxy_entry_swallows_non_serializable_extra(tmp_path):
    proxy_path = str(tmp_path / 'p.jsonl')
    with patch.object(logging_utils.ctx, 'log', MagicMock(), create=True):
        log_proxy_entry(proxy_path, 'warn', 'm', payload={'b': b'binary'})
    # No write should have occurred (json.dumps failed before os.open).
    assert not Path(proxy_path).exists()

def test_log_proxy_entry_extra_does_not_silently_shadow_reserved_keys(tmp_path):
    # Currently entry.update(extra) at logging_utils.py:37 allows `extra`
    # to overwrite `timestamp`/`level`/`message`. Pin the chosen behavior
    # (raise, ignore, or namespace) so future refactors don't drift.
    ...
```

The third test is the most valuable: it surfaces the existing footgun at `logging_utils.py:37` where `**extra` silently shadows the three reserved keys the function itself sets. Pair it with whatever decision the team makes (likely: filter reserved keys before update, or nest extra under one key).

### PR DESCRIPTION
**Pull request by seven332**
## Summary

- route network and proxy JSONL logging through a shared best-effort writer
- serialize log entries before opening the destination file so serialization failures do not leave empty files
- keep proxy log reserved fields owned by the logger while preserving top-level extra fields
- cover missing-path, directory-path, non-serializable payload, and reserved-field regressions

## Tests

- `PYTHONPATH=/tmp/vm0-mitm-addon-deps /tmp/vm0-mitm-addon-deps/bin/ruff check src/logging_utils.py tests/test_utils.py tests/test_registry.py`
- `PYTHONPATH=/tmp/vm0-mitm-addon-deps /usr/bin/python3 -m pytest tests/test_utils.py tests/test_registry.py`
- `PYTHONPATH=/tmp/vm0-mitm-addon-deps /usr/bin/python3 -m pytest -q`
- `git diff --check`

Closes #14371

### CODE REVIEW

File: crates/runner/mitm-addon/tests/test_utils.py [hunk 1]
```diff
@@ -1,9 +1,12 @@
 """Tests for URL and logging utility functions."""
 
 import json
+from pathlib import Path
+from unittest.mock import MagicMock, patch
 
 import pytest
 
+import logging_utils
```
[the remark this answers is not in the document]
Comment (seven332) [reply crates/runner/mitm-addon/tests/test_utils.py:9]: Fixed in 53bbb8616b by using only the module import in test_utils.py and qualifying log_proxy_entry calls through logging_utils.

**Comment by seven332** [comment]
## Code Review: PR #14388

### Summary

Reviewed the mitm-addon logging changes that centralize best-effort JSONL writes, serialize before opening files, protect proxy-log reserved fields, and add regression coverage for failure/no-op paths.

### Key Findings

#### Critical Issues (P0)

None.

#### High Priority (P1)

None.

### Testing Review

#### Coverage

- fix: logging failures should warn and never propagate -> covered in `crates/runner/mitm-addon/tests/test_utils.py` and `crates/runner/mitm-addon/tests/test_registry.py`.
- fix: serialization failures should not create empty log files -> covered for proxy and network log helpers.
- fix: proxy-log reserved fields should remain logger-owned while dynamic extras stay top-level -> covered in `test_extra_cannot_override_reserved_fields`.
- edge case: empty paths stay true no-ops even with non-serializable payloads -> covered in both helper test groups.

#### Convention Compliance

No violations found. The tests use real `tmp_path` filesystem paths and patch only the mitmproxy `ctx.log` side effect. No filesystem mocks, internal-code mocks, fake timers, sleeps, or implementation-only assertions were added.

#### Testing Verdict: Adequate

### Bad Smell Analysis

- Mock violations: 0
- Defensive try/catch violations: 0
- Dynamic imports: 0
- Lint/type suppressions: 0
- Artificial delays or fake timers: 0
- Security concerns: 0

The shared broad catch in `_write_jsonl_entry` is intentional for this best-effort logging boundary and replaces duplicated catch blocks. The public helper behavior is covered through exported helper calls rather than private-helper tests.

### Recommendations

No changes requested.

### Verdict

LGTM.

---
*Full review details: `codereviews/20260521/`*
*Testing standards: `docs/testing.md` and `docs/testing/mitm-addon-testing.md`*
